plugins/transport-grpc/src/main/java/org/opensearch/transport/grpc/Netty4GrpcServerTransport.java

```diff
@@ -115,14 +115,29 @@ public class Netty4GrpcServerTransport extends NetworkPlugin.AuxTransport {
         Setting.Property.NodeScope
     );
 
-    private final Settings settings;
+    /**
+     * Port range on which servers bind.
+     */
+    protected PortsRange port;
+
+    /**
+     * Port settings are set using the transport type, in this case GRPC_TRANSPORT_SETTING_KEY.
+     * Child classes have distinct transport type keys and need to override these settings.
+     */
+    protected String portSettingKey = SETTING_GRPC_PORT.getKey();
```

Comment:
Let's make it final and initialize in the constructor:

```suggestion
    private final String portSettingKey;
```

Reply:
Moved initialization for this var into constructor. Protected as it's also set in `SecureNetty4GrpcServerTransport` since secure/non-secure transports have different settings for configuring ports. Let me know what you think of this pattern.